src/Init/Data/Iterators/ToIterator.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+module
+
+prelude
+import Init.Data.Iterators.Consumers
+
+/-!
+This module provides the typeclass `ToIterator`, which is implemented by types that can be
+converted into iterators.
+-/
+
+open Std.Iterators
+
+namespace Std.Slice
```

Comment:
Double-check the namespace? It's a bit weird, I don't really understand whether `ToIterator` is supposed to be this general-purpose concept or wheter it is specific to slices.